internal/cmd/cluster/process.go

```diff
@@ -1,124 +1,53 @@
 package cluster
 
 import (
-	"bytes"
-	"encoding/json"
-	"errors"
-	"fmt"
 	"io"
 	"os"
 
 	"github.com/urfave/cli/v2"
-	process "github.com/wetware/ww/pkg/process/client"
 )
 
-const (
-	_module = "module"
-	_func   = "function"
-)
-
-var runError = errors.New("Run failed.")
-
 func run() *cli.Command {
 	return &cli.Command{
-		Name:  "run",
-		Usage: "run a WASM module",
-		Flags: []cli.Flag{
-			&cli.StringFlag{
-				Name:     _module,
-				Aliases:  []string{"m"},
-				Usage:    "path to the file containing compiled WASM module",
-				Required: true,
-			},
-			&cli.StringFlag{
-				Name:     _func,
-				Aliases:  []string{"f"},
-				Usage:    "name of the function to run within the WASM module",
-				Required: true,
-			},
-			&boolFlag,
-		},
-		Before: setup(),
-		After:  teardown(),
-		Action: runAction(),
+		Name:      "run",
+		Usage:     "compile and run a WASM module",
+		ArgsUsage: "<path> (defaults to stdin)",
+		Before:    setup(),
+		After:     teardown(),
+		Action:    runAction(),
 	}
 }
 
 func runAction() cli.ActionFunc {
 	return func(c *cli.Context) error {
-
-		var stdin io.Reader = os.Stdin
-		var stdout, stderr io.Writer
-
 		ctx := c.Context
+
 		// Load the name of the entry function and the WASM file containing the module to run
-		entryFunction := c.String(_func)
-		binary, err := os.ReadFile(c.String(_module))
+		src, err := bytecode(c)
 		if err != nil {
 			return err
 		}
 
 		// Obtain an executor and spawn a process
 		executor, release := node.Executor(ctx)
 		defer release()
-		proc := process.MakeProcess(ctx, logger, executor, binary, entryFunction)
-		defer proc.Close(ctx)
-
-		// Select the output
-		if c.Bool(_json) {
-			stdout = new(bytes.Buffer)
-			stderr = new(bytes.Buffer)
-		} else {
-			stdout = os.Stdout
-			stderr = os.Stderr
-		}
 
-		// Run the process
-		errs := proc.Run(ctx, stdin, stdout, stderr)
+		proc, release := executor.Spawn(ctx, src)
```

Comment:
Aren't we missing the entry function parameter? Both here and when calling the capability

Reply:
Ah yes, see the `process.capnp` schema file.  I specified a default value for that parameter.  Right now there is no way to override that default (for simplicity), but we can add it in later if it becomes necessary.